packages/apps/webconnector/src/services/webConnector.ts

```diff
@@ -0,0 +1,269 @@
+import fs from "fs";
+import axios from "axios";
+import { Readable } from "stream";
+import {
+  OAuthService,
+  Organisation,
+  OrganisationService,
+} from "@ocular/ocular";
+import {
+  IndexableDocument,
+  TransactionBaseService,
+  Logger,
+  AppNameDefinitions,
+  DocType,
+} from "@ocular/types";
+import { ConfigModule } from "@ocular/ocular/src/types";
+import puppeteer, { Browser, Page } from "puppeteer";
+
+interface metadataLink {
+  location: string;
+  status: string;
+}
+
+export default class webConnectorService extends TransactionBaseService {
+  protected oauthService_: OAuthService;
+  protected logger_: Logger;
+  protected container_: ConfigModule;
+  protected organisationService: OrganisationService;
+
+  constructor(container) {
+    super(arguments[0]);
+    this.oauthService_ = container.oauthService;
+    this.logger_ = container.logger;
+    this.container_ = container;
+  }
+
+  async getWebConnectorData(
+    org: Organisation,
+    base_url: string,
+    link_id: string
+  ) {
+    return Readable.from(this.crawlWebConnectorData(org, base_url, link_id));
+  }
+
+  async *crawlWebConnectorData(
+    org: Organisation,
+    base_url: string,
+    link_id: string
+  ): AsyncGenerator<IndexableDocument[]> {
+    this.logger_.info(
+      `Starting oculation of webConnector for ${org.id} organisation`
+    );
+
+    const oauth = await this.oauthService_.retrieve({
+      id: org.id,
+      app_name: AppNameDefinitions.WEBCONNECTOR,
+    });
+
+    if (!oauth) {
+      this.logger_.error(
+        `No WebConnector OAuth found for ${org.id} organisation`
+      );
+      return;
+    }
+
+    try {
+      const pageArray = await this.crawl(base_url);
+      let documents: IndexableDocument[] = [];
+      for (const page of pageArray) {
+        if (page.text) {
+          const pageDocument: IndexableDocument = {
+            id: page.location,
+            organisationId: org.id,
+            title: page.text,
```

Comment:
Why is the title = page.text here? I think title should be the first heading in the page. Or least the first 50 characters of page.text!